src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -223,6 +229,10 @@ protected override void OnSysCall(uint method)
 
         protected override void PreExecuteInstruction()
         {
+            ExecutionContextState state = CurrentContext.GetState<ExecutionContextState>();
+            if (!state.TuringComplete)
+                if (CurrentContext.CurrentInstruction.OpCode >= OpCode.JMP && CurrentContext.CurrentInstruction.OpCode <= OpCode.CALLA)
```

Comment:
Nice, prevents all jumps from JMP to CALLA.

Reply:
Shall we block SYSCALLs (or some of them) here?

Reply:
I think we should inspect JMP payload and only block those with negative offsets, preserving `if-then-else` structure and removing loops... this will make Verification much more useful, without ifs it's hard (or impossible) to code anything apart from standard pubkey checkings (not even multisig!).

Reply:
Update: this proposed if-then-else blocking is ok for InvocationScript.. but I think VerificationScript should also be limited (but allowing ifs).
In this sense, it would be fine for me to limit both Invocaton and Verification scripts with a variant of this that allowed ifs and blocked loops, as this would be easier to understand and already protect network from undesired spam operations.